kong/dao/factory.lua

```diff
@@ -417,6 +417,14 @@ local function migrate(self, identifier, migrations_modules, cur_migrations, on_
       err = migration.up(self.db, self.kong_config, self)
     end
 
+    if err and migration.ignore_error then
+      if type(migration.ignore_error) == "boolean" or
+         (type(migration.ignore_error) == "string" and
+          string.find(tostring(err), migration.ignore_error)) then
```

Comment:
I suggest chaning this to `string.find(tostring(err), migration.ignore_error, 1, true)` so that it does a plain-text match, just so this doesn't fail mysteriously someday in the future when we have a message that happens to have a hyphen.